app/actions/user.ts

```diff
@@ -1,171 +1,153 @@
-import { providers } from "ethers";
+import { providers, ethers } from "ethers";
 import { Thunk } from "state";
 
-import { addresses } from "@klimadao/lib/constants";
 import {
   formatUnits,
-  trimStringDecimals,
   getContract,
+  createContractsObject,
+  getAllowance,
+  getTokensFromSpender,
+  getTokenDecimals,
 } from "@klimadao/lib/utils";
-import {
-  setBalance,
-  setExerciseAllowance,
-  setStakeAllowance,
-  setWrapAllowance,
-  setDomains,
-} from "state/user";
+import { AllowancesFormatted } from "@klimadao/lib/types/allowances";
+import { setBalance, updateAllowances, setDomains } from "state/user";
 
 import { getKns, getEns } from "./utils";
 
+const assets = [
+  "bct",
+  "nct",
+  "mco2",
+  "ubo",
+  "nbo",
+  "usdc",
+  "klima",
+  "sklima",
+  "wsklima",
+  "pklima",
+] as const;
+type Asset = typeof assets[number];
+
+const spenders = [
+  "staking_helper",
+  "staking",
+  "wsklima",
+  "pklima_exercise",
+] as const;
+
+type TokenValueFormatted = {
+  [K in Asset]: string;
+};
+
+const getBalance = async (params: {
+  token: Asset;
+  contract: ethers.Contract;
```

Comment:
would be cool if this was smart enough to find the correct contract on it's own :)

future enhancement?

Reply:
Yeah, let´s discuss a smart way on how to init all contracts in one place in lib
so that both app and site can have access to it